### ISSUE TITLE
Fall back to the original link address when displaying a profile page is not possible

### ISSUE THREAD

**Issue by nickludlam** [TheBLVD/mammoth#191]
### Is your feature request related to a problem? Please describe.

Earlier today I see people excitedly posting about Threads gaining a test of interoperability with ActivityPub and Mastodon. In particular I see this Toot reposted to my timeline:

https://social.coop/@J12t/111577315626579051

This contains a mention of user mosseri.  The original toot content contains:

```
<a href="https://www.threads.net/@mosseri/" class="u-url mention" rel="nofollow noopener noreferrer" target="_blank">@<span>mosseri</span></a>
```

This link is to the address https://www.threads.net/@mosseri/ which resolves to a valid threads.net profile page.

Now this HTML is stripped out by `PostCardModel.formattedPostText()`, and links are handled separately by [onMentionPress()](https://github.com/TheBLVD/mammoth/blob/f4981be3208badb84ec6abe8520b3bd0b61a42fa/Mammoth/Utils/PostActions.swift#L699).

This then invokes `ProfileViewController()`, and the mention data is:

```
mention.acct=mosseri@threads.net
mention.username=mosseri
mention.url=https://www.threads.net/@mosseri/
```

So we still have a valid URL at this point. This then invokes `ProfileViewController()` which attempts to call the [Mastodon account lookup endpoint](https://docs.joinmastodon.org/methods/accounts/#lookup).

This then expects the threads.net to support a GET request to `/api/v1/accounts/lookup` which it currently does not and I'm not sure ever will, so it gets a 404 and fails to look the user up.  This gives the user an unrecoverable UIAlert, which makes it look like there is a hard failure state when there is a perfectly valid web page it could fall back to displaying.

### Describe the solution you'd like.

I don't know the right answer for what to display when you tap on a username that it isn't referencing a native Mastodon user, but I think it's poor behaviour to replace a valid hyperlink with a profile page subview that will never work.  It violates the principle of least surprise.

In this case I'd like it to fall back to showing you the webpage in a webview at the very least, when the profile lookup fails.

### Describe alternatives you've considered.

I assume it would be possible to adapt the Profile Viewer system to work with a range of user sources, and maintain the user expectation that tapping on any username brings you to their profile page. This sounds like a larger rewrite.

### Additional context.

I am not an expert on ActivityPub or WebFinger so there may well be more elegant ways of processing the lookups by working with the lower level protocols more directly.

**Comment by derspyy** [comment]
for this specific issue, if the lookup was done on the local instance (i.e. moth.social) instead of threads it'd return the profile properly

**Comment by derspyy** [comment]
> I don't know the right answer for what to display when you tap on a username that it isn't referencing a native Mastodon user

it should open the profile normally, because the profile exists on the mastodon server even if it's not a mastodon profile

**Comment by derspyy** [comment]
i mean... just click the user mention on the web ui. it just opens the profile

**Comment by bnolens** [comment]
Thanks for reporting this @derspyy. I propose we solve this by loading the profile through the user's instance instead, when it's a Threads profile. https://github.com/TheBLVD/mammoth/pull/203

**Comment by derspyy** [comment]
> Thanks for reporting this @derspyy. I propose we solve this by loading the profile through the user's instance instead, when it's a Threads profile. #203
@bnolens 

shouldn't this already have been part of the fallback we implemented ages ago? i feel like there may be something wrong with that...

**Comment by bnolens** [comment]
> > Thanks for reporting this @derspyy. I propose we solve this by loading the profile through the user's instance instead, when it's a Threads profile. #203
> > @bnolens
> 
> shouldn't this already have been part of the fallback we implemented ages ago? i feel like there may be something wrong with that...

When tapping on a `@mention` and opening a profile screen, we rely on the mentions object in the Status. There's no fallback logic in place here.

**Comment by derspyy** [comment]
ohhh, okay. 
so maybe setting up a fallback is preferable since it'd maintain the same logic as loading the profile from other places. 
because if this fails with threads, it'll fail with every server that's not mastodon.

**Comment by bnolens** [comment]
> ohhh, okay. so maybe setting up a fallback is preferable since it'd maintain the same logic as loading the profile from other places. because if this fails with threads, it'll fail with every server that's not mastodon.

Good point @derspyy. I've added a fallback for non-mastodon instances

**Comment by bvon** [comment]
Fixed in 2.1!
